### PR TITLE
BF: Protect against multiple JSON records being read at once

### PR DESCRIPTION
**Pull request by mih**
Fixes gh-4236

### CODE REVIEW

File: datalad/support/annexrepo.py [hunk 1]
```diff
@@ -3524,15 +3524,20 @@ def pipe_data_received(self, fd, data):
...
-                lgr.error('Received undecodable JSON output: %s', data)
-            return
+        for line in data.splitlines():
+            try:
+                j = json_loads(line)
+            except Exception:
+                # TODO turn this into an error result, or put the exception
+                # onto the result future -- needs more thought
+                if line.strip():
+                    print("FUCKINGDECODE", data)
```
Comment (yarikoptic) [inline datalad/support/annexrepo.py]: leftover

**Comment by mih** [comment]
Those are rather curious CI failures. Even more so, as they are Windows only.

**Comment by kyleam** [comment]
> Those are rather curious CI failures. Even more so, as they are Windows only.

Comparing several PR that now show these failures to recent PRs that don't, the main difference I'm spotting is that the failing ones seem to be using Git 2.25.1, but all the recent passing ones I've been able to find use 2.25.*0*.  Seems pretty unlikely that that would matter, though.

**Comment by kyleam** [comment]
> the main difference I'm spotting is that the failing ones seem to be using Git 2.25.1

I tried tweaking some of the config variables specific to Git for Windows, in particular `core.fscache` based on it being mentioned in 2.25.1.windows.1 release notes.  That didn't resolve these failures.

**Comment by mih** [comment]
Given that this PR fixes a real issue that affects real systems I will merge this now and worry about the GitHub Windows CI oddities later/separately. Thx @kyleam for looking into this!

**Pull request merged**